src/odemis/driver/actuator.py

```diff
@@ -1477,12 +1491,22 @@ def _readPositionfromChildren(self):
         check if the children axes positions correspond to any allowed combined axis position
         :return: position key in position dict for child axis
         """
+        # get current children positions
         pos_children_cur = [c.position.value[ca] for c, ca in zip(self._children, self._caxes_map)]
 
-        # check whether children positions are in consistency with positions allowed
+        # check whether current children positions are in consistency with children positions allowed
         for pos_key, pos_children in self._positions.items():
-            for ap, cp, atol in zip(pos_children_cur, pos_children, self._atol):
-                if isinstance(cp, numbers.Real) and not util.almost_equal(ap, cp, atol=atol, rtol=0):
+            # ap: current children position, cp: children position possible
+            for ap, cp, atol, cyl in zip(pos_children_cur, pos_children, self._atol, self._cycle):
+                # handle positions close to cycle and thus also close to zero
+                # and within tolerance so util.almost_equal compare the correct values
+                if cyl is None:
+                    dist = abs(ap - cp)
+                else:
+                    dist = min((ap - cp) % cyl, (cp - ap) % cyl)
+                # if the first pos matches, check also the next. Else don't check any further
```

Comment:
That's already explained in the comment next to the `else:`. No need to repeat it here.